File: CBox/model/Headline.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel;

namespace cbox.model
{
    public class Headline
    {
        public string Title { get; set; }
        public BindingList<int> ActionIdents { get; set; }

        public Headline()
        {
            ActionIdents = new BindingList<int>();
            Title = "New headline";
        }
    }
}

File: CBoxClient/lib/model/Headline.ts
module cbox {

    export class Headline {

        title: string;
        action_idents: number[] = [];

        public static fromObject(obj: {}): Headline {
            var h = new Headline();

            h.title = obj['Title'];
            h.action_idents = obj['ActionIdents'].map((i) => { return parseInt(i) } );

            return h;
        }

    }

}